### PR TITLE
Xpetra: Fixes to Xpetra::TpetraVector

### PR DESCRIPTION
**Pull request by csiefer2**
Auto-PR for SHA 7b030c6

### CODE REVIEW

**Comment by prwolfe** [comment]
This failure was my fault - I am setting a retest flag

**Comment by csiefer2** [comment]
Roger that @prwolfe

File: packages/xpetra/src/Vector/Xpetra_TpetraVector_def.hpp [hunk 1]
```diff
@@ -115,11 +115,11 @@ void TpetraVector<Scalar,LocalOrdinal,GlobalOrdinal,Node>::describe(Teuchos::Fan
 { XPETRA_MONITOR("TpetraVector<Scalar,LocalOrdinal,GlobalOrdinal,Node>::describe"); getTpetra_Vector()->describe(out, verbLevel); }
 
 template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
-Scalar TpetraVector<Scalar,LocalOrdinal,GlobalOrdinal,Node>::dot(const Vector &a) const
+Scalar TpetraVector<Scalar,LocalOrdinal,GlobalOrdinal,Node>::dot(const Vector<Scalar,LocalOrdinal, GlobalOrdinal, Node > &a) const
 { XPETRA_MONITOR("TpetraVector<Scalar,LocalOrdinal,GlobalOrdinal,Node>::dot"); return getTpetra_Vector()->dot(*toTpetra(a)); }
 
 template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
-typename Teuchos::ScalarTraits< Scalar >::magnitudeType TpetraVector<Scalar,LocalOrdinal,GlobalOrdinal,Node>::normWeighted(const Vector &weights) const
+typename Teuchos::ScalarTraits< Scalar >::magnitudeType TpetraVector<Scalar,LocalOrdinal,GlobalOrdinal,Node>::normWeighted(const Vector<Scalar,LocalOrdinal, GlobalOrdinal, Node > &weights) const
```
Comment (mhoemmen) [inline packages/xpetra/src/Vector/Xpetra_TpetraVector_def.hpp]: Tpetra has deprecated this.  Does MueLu actually need it still?  (Would be good to know that before Tpetra removes it entirely ;-) .)

Comment (csiefer2) [reply packages/xpetra/src/Vector/Xpetra_TpetraVector_def.hpp]: Grep says we don't use it.

Comment (mhoemmen) [reply packages/xpetra/src/Vector/Xpetra_TpetraVector_def.hpp]: @csiefer2 I'm not sure anyone would consider most of Xpetra a public interface, so I think it's fair game to purge or at least deprecate.

Comment (csiefer2) [reply packages/xpetra/src/Vector/Xpetra_TpetraVector_def.hpp]: I'm totally OK with purging it when Tpetra does.

**Pull request merged**